content/en/observability_pipelines/processors/remap_ocsf/custom_configuration.md

```diff
@@ -0,0 +1,369 @@
+|----------|----------|-------------------------------------------------------------------------------------------------|
+| version  | Yes      | The target OCSF schema version.                                                                 |
+| class    | Yes      | The name of the OCSF event class. The event class ID, category name, and category ID are derived from this. |
+| profiles | Yes      | An array of profile names that are added to the event class schema.                             |
+
+
+{{< tabs >}}
+{{% tab "YAML example" %}}
+
+```yaml
+version: 1
+metadata:
+  version: 1.1.0
+  class: Network Activity
+  profiles: [container, host, security_control]
+preprocess: []
+mapping: []
+
+```
+{{% /tab %}}
+{{% tab "JSON example" %}}
+
+```json
+{
+  "version": 1,
+  "metadata": {
+    "version": "1.1.0",
+    "class": "Network Activity",
+    "profiles": ["container", "host", "security_control"]
+  },
+  "preprocess": [],
+  "mapping": []
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+
+### Preprocessors
+
+Preprocessors rework the data to allow field-to-field mappings. Each entry in the `preprocess` section is an object consisting of a `function` name (required) and parameters associated with that function.
+
+#### `parse_csv`
+
+The `parse_csv` preprocessor:
+
+1. Extracts a `source` field.
+1. Parses `source` as CSV, maps the array of values to an object based on a list of field names in `columns`.
+1. Inserts it at `dest`.
+
+Fields with a corresponding `null` column name are dropped. One of the field names may have a wildcard `*` in which case it is assigned a string containing all the text from the fields that remain after those before and after have been mapped.
+
+{{< tabs >}}
+{{% tab "YAML example" %}}
+
+```yaml
+preprocess:
+- function: parse_csv
+  source: message
+  dest: .
+  skip_empty: true
+  - null
+  - receive_time
+  - serial
+  - type
+  - subtype
+  - null
+  - 'wildcard*'
+  - time_generated
+```
+
+{{% /tab %}}
+
+{{% tab "JSON example" %}}
+
+```json
+{
+  "preprocess": [
+    {
+      "function": "parse_csv",
+      "source": "message",
+      "dest": ".",
+      "skip_empty": true,
+      "columns": [
+        null,
+        "receive_time",
+        "serial",
+        "type",
+        "subtype",
+        null,
+        "wildcard*",
+        "time_generated"
+      ]
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+
+In the examples above, the preprocessor expects an array object with seven or more columns. Since the first and sixth columns are `null`, they are dropped. The other six columns are mapped to:
+- `receive_time`
+- `serial`
+- `type`
+- `subtype`
+- `time_generated`
+- `wildcard`, which is set to the remaining columns of the array.
+
+### Mapping
+
+Each mapping entry consists of a `dest` path naming the destination field in the OCSF event and either a `source` path that names a field in the source event or a `value` that contains a literal constant to insert at that destination.
```

Comment:
I'm wondering if there's a way to show this in an example. I don't really understand what's going on here just from the text.

Reply:
In the trivial two-item example below, if the source contains:
```json
{
  "authenticationContext": {
    "issuer": {
      "id": "something"
    }
  }
}
```

then it would produce the following fields in the OCSF output:
```json
{
  "activity_id": 1,
  "activity_name": "Create",
  "actor": {
    "session": {
      "issuer": "something"
    }
  }
}
```
(along with the rest of the bits needed to support the OCSF output; the `1` above is representative)